### PR TITLE
refactor: useDetectDevice hook

### LINKED ISSUE #1045: Use common listener for device type
Currently there are 3 listeners to detect device type across Header, SearchResults and HomePage components. 

Would be good to explore a way to unify these 3 listeners by creating a shared context for device type.

### PR DESCRIPTION
**Pull request by nigel0511**
Fixes #1045 

replace pull request #1448 

1. extract detect device functions from Header, SearchResults and AgencyHomePage component.
2. create useDetectDevice hook in hooks file and create getCurrentDeviceSize() for replicated code.
3. replace detect device functions with useDetectDevice.

### CODE REVIEW

File: client/src/hooks/useDetectDevice.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from 'react'
+
+export const useDetectDevice = () => {
+  const device = {
+    mobile: 'mobile',
+    tablet: 'tablet',
+    desktop: 'desktop',
+  }
```
Comment (LoneRifle) [inline client/src/hooks/useDetectDevice.ts]: Consider making this an enum exported from this module. Perhaps something like..

```ts
export enum Device {
  Mobile = 'mobile',
  // ...
}

File: client/src/pages/AgencyHomePage/AgencyHomePage.component.tsx [hunk 2]
```diff
@@ -238,7 +212,7 @@ const AgencyHomePage = (): JSX.Element => {
         ? homePageDefaultView // no topics param key -> default homepage
         : !topicQueried
         ? homePageAllQuestionsView // topics param key + no topic queried -> all questions
-        : deviceType === device.desktop // specific topic selected
+        : deviceType === 'desktop' // specific topic selected
```
Comment (LoneRifle) [inline client/src/pages/AgencyHomePage/AgencyHomePage.component.tsx]: we shouldn't be hardcoding string literals here, as we then lose the constraints that limit the comparison of `deviceType` to specific values. import the proposed enum mentioned elsewhere, and use those values instead.

**Review by LoneRifle — CHANGES_REQUESTED** [review]
Have a look at the things I've highlighted

File: client/src/pages/AgencyHomePage/AgencyHomePage.component.tsx (hunk 2 above)
Comment (nigel0511) [reply client/src/pages/AgencyHomePage/AgencyHomePage.component.tsx]: okay, I will make the suggested change, thank you for pointing that out.

**Comment by nigel0511** [comment]
Change made:
1. created devicetype.ts under util, export enum device.
2. replace string literals with variables 

please advise if that's appropriate, open to suggestions.

**Review by LoneRifle — APPROVED** [review]

**Pull request merged**